piksi_tools/console/console.py

```diff
@@ -328,6 +349,34 @@ def update_serial():
       import traceback
       traceback.print_exc()
 
+# Make sure that SIGINT (i.e. Ctrl-C from command line) actually stops the
+# application event loop (otherwise Qt swallows KeyboardInterrupt exceptions)
+signal.signal(signal.SIGINT, signal.SIG_DFL)
+
+# Passing only a base station argument, we just want to display the
+# base station data in the console. Otherwise, continue, assuming a
+# rover connected to the serial port.
+if port is None and base:
+  device_id = get_uuid(channel, serial_id)
+  with HTTPDriver(str(device_id)) as http_driver:
+    with sbpc.Handler(sbpc.Framer(http_driver.read, None, args.verbose)) as link:
+      with s.get_logger(args.log, log_filename) as logger:
+        link.add_callback(logger)
+        log_filter = DEFAULT_LOG_LEVEL_FILTER
+        if args.initloglevel[0]:
+          log_filter = args.initloglevel[0]
+        SwiftConsole(link, args.update, log_filter, True).configure_traits()
+  try:
+    os._exit(0)
+  except:
+    pass
+elif port is None and base is None:
```

Comment:
I was thinking of:

```python
if port is None and base is None:
  sys.stderr.write("ERROR: No data source specified!")
  sys.exit(1)

if port is None and base:
  ...
```

Isolate the error condition and don't bother attaching it to the other check. Just a style and comprehension thing